rotkehlchen/tests/conftest.py

```diff
@@ -246,13 +250,26 @@ def fixture_vcr_base_dir() -> Path:
     log.debug(f'At VCR setup, {current_branch=} {root_dir=}')
 
     checkout_proc = Popen(f'cd "{root_dir}" && git fetch origin && git checkout {current_branch}', shell=True, stdout=PIPE, stderr=PIPE)  # noqa: E501
-    _, stderr = checkout_proc.communicate(timeout=SUBPROCESS_TIMEOUT)
+
+    try_fallback_branch = False
+    try:
+        _, stderr = checkout_proc.communicate(timeout=SUBPROCESS_TIMEOUT)
+    except TimeoutExpired:
+        stderr = b''
+        try_fallback_branch = True
+
     if (
-        len(stderr) != 0 and
-        b'Already on' not in stderr and
-        b'Switched to' not in stderr
+        try_fallback_branch is True or
+        (
+            len(stderr) != 0 and
+            b'Already on' not in stderr and
+            b'Switched to' not in stderr
+        )
     ):
         default_branch = os.environ.get('GITHUB_BASE_REF', os.environ.get('DEFAULT_VCR_BRANCH', 'develop'))  # noqa: E501
+        if len(default_branch) == 0 and 'CI' in os.environ:
```

Comment:
When is `len(default_branch) == 0`? If someone sets `GITHUB_BASE_REF` and/or `DEFAULT_VCR_BRANCH` to `''`? When can that happen?

Reply:
`GITHUB_BASE_REF` is set to `''` for jobs that don't run from a pull requests and in the CI we don't set `DEFAULT_VCR_BRANCH`. If someone set this variables to `''` locally is the user faults but in the case of the CI we need to protect against this case

Reply:
again same as above. Comment and comparison to empty string. The length check is really realy confusing here.